xarray/core/datatree_mapping.py

```diff
@@ -4,10 +4,9 @@
 import sys
```

Comment:
@TomNicholas - there are no significant changes to the way `map_over_subtree` works in this PR. As discussed today, I've opened the PR, and am happy to make changes based on conversation here.

Reply:
Okay looking at this again I can't see an obvious way to refactor it. I'm mostly just concerned that it actually makes sense to you @owenlittlejohns ?

If so then I suggest we spend the time thinking about https://github.com/pydata/xarray/issues/8949 instead

Reply:
@TomNicholas - yeah. @flamingbear and I have both spent a fair bit of time trying to grok this module. I don't claim that I've got a perfect understanding, but I feel fairly good on the whole. (The conversation yesterday that led to #8949 helped a bit, too)

Reply:
Okay great. So I guess we merge this??

FYI if you want to chat about any of this over zoom outside of the weekly meeting I would be happy to.

Reply:
I think we're good to merge!